src/Agent.Worker/TaskManager.cs

```diff
@@ -120,12 +145,36 @@ private async Task DownloadAsync(IExecutionContext executionContext, Pipelines.T
             // first check to see if we already have the task
             string destDirectory = GetDirectory(task);
             Trace.Info($"Ensuring task exists: ID '{task.Id}', version '{task.Version}', name '{task.Name}', directory '{destDirectory}'.");
-            if (File.Exists(destDirectory + ".completed"))
+
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
+            AgentSettings settings = configurationStore.GetSettings();
+            Boolean signingEnabled = !String.IsNullOrEmpty(settings.Fingerprint);
```

Comment:
Are we not tracking "signingEnabled" per task? It seems like this could be a problem if you only want to sign a few tasks, but I haven't thought thru it.

Reply:
Correct. The requirements for highly regulated environments is that all tasks need to be signed when running. We may consider a mixed-mode in the future.